src/controllers/keystore/keystore.ts

```diff
@@ -439,9 +446,61 @@ export class KeystoreController extends EventEmitter {
 
     // Set up the cipher
     const counter = new aes.Counter(this.#mainKey!.iv) // TS compiler fails to detect we check for null above
-    const aesCtr = new aes.ModeOfOperation.ctr(this.#mainKey!.key, counter) // TS compiler fails to detect we check for null above
+    const aesCtr = new aes.ModeOfOperation.ctr(this.#mainKey!.key, counter) // TS compiler fails to detect we check for null above\
+    return hexlify(aesCtr.encrypt(new TextEncoder().encode(seed)))
+  }
+
+  async addSeedToTemp({ seed, hdPathTemplate }: KeystoreSeed) {
```

Comment:
I'd try to be a bit more protective here (veggly validating if `seed` and `hdPathTemplate` incoming are defined or are in the expected format), because if for some reason this method gets invoked by a broken front-end, this might brick the KeyStore.

Reply:
Hey, Kalo, thanks :)  
For the seed we have validation - I moved it to `#getEncryptedSeed`, which is called in this method. It does:
```
if (!Mnemonic.isValidMnemonic(seed)) {
      throw new EmittableError({
        message: 'You are trying to store an invalid seed phrase.',
        level: 'major',
        error: new Error('keystore: trying to add an invalid seed phrase')
      })
    }
```

i only don't know how to validate `hdPathTemplate`. Suggestions?

Reply:
Yep, `DERIVATION_OPTIONS.some((o) => o.value === hdPathTemplate)`